saleor/product/models.py

```diff
@@ -462,6 +466,11 @@ def get_first_image(self) -> "ProductImage":
         images = list(self.images.all())
         return images[0] if images else self.product.get_first_image()
 
+    def set_as_default(self):
```

Comment:
We should avoid this kind of utils. Each usage of this function generates 2 database queries. 
It's hard to optimize.

Reply:
This utility is used in mutation so I'm not sure if we can further optimize it. But I would move it to `product/utils` to keep the model thinner.